core/oracles/ethereum_oracle_verifier.go

```diff
@@ -180,7 +164,13 @@ func (s *EthereumOracleVerifier) checkCallEventResult(callEvent types.EthContrac
 	// Maybe TODO; can we do better than this? Might want to cancel on shutdown or something..
 	ctx := context.Background()
 	checkResult, err := s.ethEngine.CallSpec(ctx, callEvent.SpecId, callEvent.BlockHeight)
-	if err != nil {
+	if callEvent.Error != nil {
```

Comment:
I think the case where `chainEvent.Error == nil` and `err != nil` is missing here. We should fail with:
```
return fmt.Errorf("failed to execute call event spec: %w", err)
```
but instead we're going to fall down and do the bytes equal bit.

Reply:
yep, also the case where the error is nil is a bit messed up as it will log nil error with failed to execute message.  Will rework this.